src/voice-search/voice-search.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  Component,
+  Input,
+  Inject,
+  forwardRef,
+  NgZone,
+  ContentChild,
+  ElementRef,
+  TemplateRef,
+  OnInit,
+} from '@angular/core';
+
+import { connectVoiceSearch } from 'instantsearch.js/es/connectors';
+import { VoiceListeningState } from 'instantsearch.js/es/lib/voiceSearchHelper';
+import { BaseWidget } from '../base-widget';
+import { NgAisInstantSearch } from '../instantsearch/instantsearch';
+import { noop } from '../utils';
+
+type TemplateContext = {
+  status: string;
+  transcript?: string;
+  isSpeechFinal?: boolean;
```

Comment:
Why is this optional?

Reply:
This is from `VoiceListeningState` from the voice search helper. And in the voiceSearchHelper, `isSpeechFinal` is optional. It's because 

* `undefined` : not even started yet
* `false` : still listening
* `true` : speech recognition is finished

This feels natural, and I didn't have any need to merge `undefined` and `false` status. Anyway the most user will use it like `if (isSpeechFinal)`, so it will be okay.

Reply:
This is going to be non-optional once IS.js@3.5.1 is published.
The discussion behind this decision is [here](https://github.com/algolia/react-instantsearch/pull/2316#discussion_r283886705)

Reply:
I think you might have forgotten to make this non-optional after updating to @3.5.1, is that right?

Reply:
Thanks for pointing that out!
3c75a6d